File: Benchmarks/enterprise/python/flask/testcode/benchmark_test_42965.py
# SPDX-License-Identifier: Apache-2.0
from flask import redirect
import json
from flask import request


def BenchmarkTest42965():
    json_value = (request.get_json(silent=True) or {}).get('payload', '')
    data = json.loads(json_value).get('value', '')
    return redirect(str(data))

File: Benchmarks/enterprise/java/jakarta/testcode/BenchmarkTest42965.java
// SPDX-License-Identifier: Apache-2.0
import jakarta.servlet.http.HttpServletRequest;
import jakarta.servlet.http.HttpServletResponse;
import jakarta.ws.rs.*;
import jakarta.ws.rs.core.Context;
import jakarta.ws.rs.core.MediaType;
import jakarta.ws.rs.core.Response;

@Path("/")
public class BenchmarkTest42965 {

    @GET
    @Path("/BenchmarkTest42965")
    @Produces(MediaType.APPLICATION_JSON)
    public Response BenchmarkTest42965(@HeaderParam("Host") String host, @Context HttpServletRequest request, @Context HttpServletResponse response) throws Exception {
        String hostValue = host != null ? host : "";
        String data = String.join(" ", hostValue.split("\\s+"));
        if ("admin".equals(data) || "ROLE_ADMIN".equals(data)) {
            return Response.ok("{\"status\":\"ok\"}", MediaType.APPLICATION_JSON).build();
        }
        return Response.status(403).entity("forbidden").build();
    }
}
